flink-yarn/src/test/java/org/apache/flink/yarn/YarnResourceManagerTest.java

```diff
@@ -446,8 +459,8 @@ public void testStopWorker() throws Exception {
 
 				unregisterAndReleaseFuture.get();
 
-				verify(mockNMClient).stopContainerAsync(any(ContainerId.class), any(NodeId.class));
 				verify(mockResourceManagerClient).releaseAssignedContainer(any(ContainerId.class));
+				testingContainerStopFuture.get(TIMEOUT.getSize(), TIMEOUT.getUnit());
```

Comment:
This should come before the previous line, because containers are first stopped, then released.